### DISCUSSION TITLE
Creating indexes

### DISCUSSION THREAD

**Discussion by Foxsterc** [simolus3/drift#2139] [General]
Hi,

In reference to these two issues, is there a robust way of  doing this now?

https://github.com/simolus3/drift/issues/442
https://github.com/simolus3/drift/issues/447

If I create a drift file

```
import 'models/problem/problem.dart';

CREATE INDEX idx_bm ON problems (is_bm);

CREATE INDEX idx_su_cfg ON problems (setup, config);

CREATE INDEX idx_grd ON problems (grd);

CREATE INDEX idx_sby ON problems (se_id);

CREATE INDEX idx_nme ON problems (nme);
```

The resulting .g.dart file contains duplicate table names, removing this file (drift) solves the issue.

<img width="735" alt="image" src="https://user-images.githubusercontent.com/13538417/200601071-687f8dad-5b18-463e-a180-fa73531811bc.png">

If I remove the table name from tables[] in @DriftDatabase it does not duplicate the table however, it does not generate the indexes, views also don't work as it doesn't generate all of the code.

My work around is to add them here.

```
  MigrationStrategy get migration {
    return MigrationStrategy(
      onCreate: (Migrator m) async {
        await m.createAll();

        await m.createIndex(nameIndex);
      },
      beforeOpen: (details) async {
        await customStatement('PRAGMA foreign_keys = ON');
      },
    );
  }
}
```

Thanks

**Comment by simolus3** [comment]
I think these problems have been fixed, we also have tests for this case in the generator. Do you have a small (but ideally complete) repro for this case that I could take a look at? Thanks!

**Reply by Foxsterc** [reply]
Of course, I'll put something together. Thanks.

**Comment by Foxsterc** [comment]
Hi,

Apologies for the delay. The issue seems to be caused by adding a view.

[bs_database.zip](https://github.com/simolus3/drift/files/10085788/bs_database.zip)
